firestore/integration_test_internal/src/android/arena_ref_android_test.cc

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "app/src/util_android.h"
+#include "firestore/src/android/firestore_android.h"
+#include "firestore/src/jni/arena_ref.h"
+#include "firestore/src/jni/long.h"
+
+#include "firestore_integration_test_android.h"
+
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace jni {
+namespace {
+
+using ArenaRefTestAndroid = FirestoreIntegrationTest;
```

Comment:
Please add tests for calling each method with a pending exception. You may want to use ExceptionClearGuard or just manually check for exceptions. The important thing is that we don't leak memory.